### DISCUSSION TITLE
Unsure why function doesn't work

### DISCUSSION THREAD

**Discussion by codepeach** [PostgREST/postgrest#2130] [Q&A] [answered]
I'm using supabase as my database, and I am unsure why the following function does not work. It is giving me the error: 'structure of query does not match function result type'.

```
CREATE OR REPLACE FUNCTION select_domain(query_domain_url text)
RETURNS TABLE(id int, domain_name text) AS $$
BEGIN
    RETURN QUERY SELECT domains.id, domains.domain_name FROM domains
                 WHERE domains.domain_address = query_domain_url;
END;
$$ LANGUAGE plpgsql;
```

When I remove the begin and end, and change the language to SQL, it works fine. However, I can't change the language to sql as I want to add a lot more to this function, but I can't seem to get past this error.

**Comment by steve-chavez** [comment, marked as answer]
> structure of query does not match function result type

What's the full error message? It should include the `detail` error field.

Also see https://dba.stackexchange.com/questions/220683/structure-of-query-does-not-match-function-result-type
